### ISSUE TITLE
Relocation names incorrectly shown in Visual Mode

### ISSUE THREAD

**Issue by newlog** [radare/radare2#1922]
Hi again!

I've found a case in which the name of the relocated function is not correctly shown. It happens with the Lab06-02.exe binary of the Practical Malware Analysis book. Here's the problem:

```
0x0040116b    83c408       add esp, 8                                                                                                                                          
0x0040116e    6860ea0000   push 0xea60                                                                                                                                         
0x00401173    ff1500604000 call dword [section..rdata]                                                                                                                         
unk(unk) ; section_end..data+-4235265                    ; section..rdata                                                                                                   
0x00401179    33c0         xor eax, eax                          
```

The call to that address should show the name of Sleep(). This is why:
```
:> iS
[Sections]
idx=00 vaddr=0x00401000 paddr=0x00001000 sz=20480 vsz=19064 perm=-r-x name=.text
idx=01 vaddr=0x00406000 paddr=0x00006000 sz=4096 vsz=2398 perm=-r-- name=.rdata
idx=02 vaddr=0x00407000 paddr=0x00007000 sz=12288 vsz=16136 perm=-rw- name=.data

3 sections

:> pd 20 @0x406000
           ; DATA XREF from 0x00401173 (fcn.00401130)
           ;-- section..rdata:
           0x00406000    e465         in al, 0x65  ; RELOC 32 KERNEL32.dll_Sleep ; [1] va=0x00406000 pa=0x00006000 sz=4096 vsz=2398 rwx=-r-- .rdata
           0x00406002    0000         add byte [eax], al
           ;-- reloc.KERNEL32.dll_SetStdHandle_4:
           0x00406004    40           inc eax  ; RELOC 32 KERNEL32.dll_SetStdHandle
           0x00406005    6900002e6900 imul eax, dword [eax], 0x692e00  ; RELOC 32 KERNEL32.dll_GetStringTypeW
```

In many other cases it works fine. The problem might be caused because the function address is the same as the section address start.

Thanks for your time!

**Comment by radare** [comment]
we should find a way to determine which flag is more prioritary, this problem happens in ELF with symbols and sections too.

any suggestion to change that priority thing?

**Comment by newlog** [comment]
Hi,

I thought about it, but given that I don't know the r2 internals, what I've thought might be insufficient.
I don't know if an address can have multiple flags, if so, the easy approach would be to choose its flag depending on the context. 
For example, if we have an address `x` and it has two flags, one corresponding to the section name and the other to the relocation name, if we are disasseembling a `call` we would choose the relocation name.

Basically, special cases for `call`s. 

But as I said, I don't know if this can be "easily" implemented.

**Comment by Maijin** [comment]
what's the status on this ?

**Comment by radare** [comment]
Lack of tests maybe?

**Comment by newlog** [comment]
This issue is fixed!

`[0x0040116b 10% 816 Lab06-02.exe]> pd $r @ section..text+363 # 0x40116b                                                                                                         
            0x0040116b      83c408         add esp, 8                                                                                                                           
            0x0040116e      6860ea0000     push 0xea60                                                                                                                          
            0x00401173      ff1500604000   call dword [sym.imp.KERNEL32.dll_Sleep] ;[1] ; sym.imp.KERNEL32.dll_Sleep `
